io/plugins/eu.esdihumboldt.hale.io.jdbc/src/eu/esdihumboldt/hale/io/jdbc/JDBCTableCollection.java

```diff
@@ -304,8 +304,19 @@ public int size() {
 
 			return count;
 		} catch (SQLException e) {
-			log.warn("Could not determine query size", e);
-			return UNKNOWN_SIZE;
+			log.warn("Could not determine query size by count query");
+			try (Connection connection = createConnection()) {
+				Statement st = connection.createStatement();
```

Comment:
Additionally, you could add `st.setMaxRows(1)` to add a server side limit to the query (if the JDBC driver supports that).